### ISSUE TITLE
Regression: Cliphist `1:0.6.0-1` tries to open cliphist cache directory in `/` (Access Denied)

### ISSUE THREAD

**Issue by HarshNarayanJha** [sentriz/cliphist#120]
Hey, just updated cliphist to `1:0.6.0-1` from `1:0.5.0-1` on Arch Linux. Just after updating my clipboard history was all of sudden empty, and `cliphist list` showed `Permission Error` opening directory `/cliphist`.

I downgraded and my 53MB `~/.cache/cliphist/db` file was read and history was back.

Seems like [this](https://github.com/sentriz/cliphist/commit/b3d4bd49e801a0b3438074911909dadace7ee0fb) commit is the cause.

Probably these lines don't calculate the paths correctly on my system

https://github.com/sentriz/cliphist/commit/b3d4bd49e801a0b3438074911909dadace7ee0fb#diff-50dcce3f9be84815452ac1528aa375483b2965401553bfc7b24d279ed7038d22R44-R54

These types of bugs are pretty nasty since wrong command and wrong permission and your system is wiped :)

EDIT:

It's been resolved since the latest version is - not in arch repos yet. Solution is to stick to 0.5.0 for now, dont update until 0.6.1 comes

**Comment by HarshNarayanJha** [comment]
Tested, and this snippet works correctly on my machine

```go
cacheDir, err := os.UserCacheDir()
 if err != nil {
  fmt.Println("Error getting user cache directory:", err)
  return
 }
 fmt.Println("User Cache Dir:", cacheDir)

 configDir, err := os.UserConfigDir()
 if err != nil {
  fmt.Println("Error getting user config directory:", err)
  return
 }
 fmt.Println("User Config Dir:", configDir)
```

```fish
User Cache Dir: /home/user/.cache
User Config Dir: /home/user/.config
```

**Comment by jdambron** [comment]
Having the same issue after updating to version 0.6.0 on Arch Linux.

`opening db: create cache dir: mkdir /cliphist: permission denied`

**Comment by lingllqs** [comment]
I have the same problem

**Comment by tsdh** [comment]
Same issue.  I work around it by setting the XDG variables manually in my `~/.profile`.

```sh
if [ -z "${XDG_CACHE_HOME+x}" ]; then
    export XDG_CACHE_HOME=~/.cache
fi
if [ -z "${XDG_CONFIG_HOME+x}" ]; then
    export XDG_CONFIG_HOME=~/.config
fi
```

**Comment by HarshNarayanJha** [comment]
@tsdh in my case too XDG_CACHE_HOME and CACHE are empty dont know why... The current code seems correct and should not yield the / path, but the previous code would be doing that

"$XDG_CACHE_HOME/cliphist"

Since if XDG_CACHE_HOME is empty, it would evaluate to /cliphist

This is confusing

**Comment by HarshNarayanJha** [comment]
Oh! Wait it seems like this issue was closed 7 hours ago #118

It was an issue with 0.6.0, but the latest version 0.6.1 ia not on arch repos yet.
We must wait for the update

EDIT
Found the commit, it wasn't in 0.5.0 which we had until now, 0.6.0 has the issue, 0.6.1 fixes it

https://github.com/sentriz/cliphist/commit/19ef90ee35e6e3b27219d38ee43c82f52402431b

**Comment by Alsan** [comment]
Don't known why the 1:0.6.0-1 unable to get the history list when binding to hotkey in sway, just popup the fuzzel dialog with an empty list, but work as expected by calling the script in terminal. Rolled back to old 1:0.5.0-1 already.

My key binding:
```ini
bindsym $Mod+h            exec ~/.config/sway/scripts/clip_popup.sh
```

My script:
```sh
cliphist list | fuzzel -d -w 90 -l 30 -p "Select an entry to copy it to your clipboard buffer:"| cliphist decode | wl-copy
```

**Comment by sentriz** [comment]
indeed  this is fixed in 0.6.1, which is already out on Arch. can anyone confirm? 

https://archlinux.org/packages/extra/x86_64/cliphist/
https://gitlab.archlinux.org/archlinux/packaging/packages/cliphist/-/commits/main

**Comment by sentriz** [comment]
seems like this is fixed now, but please let me know if otherwise

**Comment by Alsan** [comment]
> indeed this is fixed in 0.6.1, which is already out on Arch. can anyone confirm?
> 
> https://archlinux.org/packages/extra/x86_64/cliphist/ https://gitlab.archlinux.org/archlinux/packaging/packages/cliphist/-/commits/main

verified on arch, thanks.
